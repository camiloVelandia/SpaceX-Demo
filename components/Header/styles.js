import styled from "styled-components";

export const Head = styled.header`
  display:flex;
  justify-content:space-between;
  align-items:center;
  padding:20px;
  position:fixed;
  top:0;
  left:0;
  right:0;
  background-color:black;
  z-index:10;
`;
export const Logo = styled.a`
  height: 50px;
  width: 220px;
  cursor: pointer;
  & img {
    height: 100%;
    width: 100%;
    object-fit: contain;
  }
  @media (min-width: 768px) {
  width: 280px;
    
  }
`;
export const List = styled.ul`
  display: none;
  padding: 0;
  margin: 0;
  list-style: none;
  @media (min-width: 768px) {
  display: flex;
    
  }
`;
export const MenuContainer = styled.ul`
  display: block;
  
  @media (min-width: 768px) {
    display: none;
  }
`;
export const Item = styled.li`
  & a {
    color: white;
    text-decoration: none;
    margin-right: 30px;
    position:relative;
    padding-bottom:6px;
    padding-right:5px;
    &:after {
      content: "";
      border-bottom: 2px solid white;
      width:0;
      transition: .6s ease-out;
      position:absolute;
      bottom: 0;
      left:0;
    }
    &:hover {
      &:after {
        content: "";
        transition: .6s ease-in;
        width:100%
      }
    }
  }
`;