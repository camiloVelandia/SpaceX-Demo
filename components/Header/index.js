import React, {useState} from 'react';
import Link from 'next/link'
import {useRouter} from 'next/router'
import { slide as Menu } from "react-burger-menu";

import { Head, Logo, List, Item, MenuContainer } from "./styles";

const Header = () => {

  const [keyword, setkeyword] = useState('')
  const router = useRouter()

  const handleSubmit= e =>{
    // e.preventDefault()
    router.push(`/search/${keyword}`)
  }
  const handleChange= e =>{
    setkeyword(e.target.value)
  }


  return (
    <Head>
      <Link href="/">
        <Logo>
          <img src="/images/logo.png" />
        </Logo>
      </Link>
      <nav>
        <List>
          <Item>
            <Link href="/rockets">
              <a>Rockets</a>
            </Link>
          </Item>
          <Item>
            <Link href="/launches">
              <a>Launches</a>
            </Link>
          </Item>
          <Item>
            <Link href="/capsules">
              <a>Dragons</a>
            </Link>
          </Item>
          <Item>
            <Link href="/stats">
              <a>Stats</a>
            </Link>
          </Item>
          <Item>
            <Link href="/about">
              <a>About</a>
            </Link>
          </Item>
          <Item>
            <form onSubmit={handleSubmit}>
              <input
                type="text"
                placeholder="search..."
                onChange={handleChange}
                value={keyword}
              />
            </form>
          </Item>
        </List>
      </nav>
      <MenuContainer>
        <Menu right width={"100%"} s>
          <Item>
            <Link href="/rockets">
              <a className="menu-item">Rockets</a>
            </Link>
          </Item>
          <Item>
            <Link href="/launches">
              <a className="menu-item">Launches</a>
            </Link>
          </Item>
          <Item>
            <Link href="/capsules">
              <a className="menu-item">Capsules</a>
            </Link>
          </Item>
          <Item>
            <Link href="/stats">
              <a className="menu-item">Stats</a>
            </Link>
          </Item>
          <Item>
            <Link href="/about">
              <a className="menu-item">About</a>
            </Link>
          </Item>
        </Menu>
      </MenuContainer>
    </Head>
  );
};



export default Header;