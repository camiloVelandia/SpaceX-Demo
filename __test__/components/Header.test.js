import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import Header from "../../components/Header";

describe("<Header/>", () => {
  const header = mount(<Header />);
  test("render del componente header", () => {
    expect(header.length).toEqual(1);
  });

});

describe("Header snaaapshot", () => {
  test("Comprobar ui del header", () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});
