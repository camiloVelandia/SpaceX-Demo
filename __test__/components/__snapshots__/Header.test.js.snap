// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snaaapshot Comprobar ui del header 1`] = `
<header
  className="styles__Head-sc-2csmix-0 iofGAm"
>
  <a
    className="styles__Logo-sc-2csmix-1 fXDohn"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      src="/images/logo.png"
    />
  </a>
  <nav>
    <ul
      className="styles__List-sc-2csmix-2 laJKla"
    >
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <a
          href="/rockets"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <a
          href="/launches"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Launches
        </a>
      </li>
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <a
          href="/capsules"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Dragons
        </a>
      </li>
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <a
          href="/stats"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Stats
        </a>
      </li>
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <a
          href="/about"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          About
        </a>
      </li>
      <li
        className="styles__Item-sc-2csmix-4 jVPdVz"
      >
        <form
          onSubmit={[Function]}
        >
          <input
            onChange={[Function]}
            placeholder="search..."
            type="text"
            value=""
          />
        </form>
      </li>
    </ul>
  </nav>
  <ul
    className="styles__MenuContainer-sc-2csmix-3 fPLzOz"
  >
    <div>
      <div
        className="bm-overlay"
        onClick={[Function]}
        style={
          Object {
            "MozTransform": "translate3d(100%, 0, 0)",
            "MsTransform": "translate3d(100%, 0, 0)",
            "OTransform": "translate3d(100%, 0, 0)",
            "WebkitTransform": "translate3d(100%, 0, 0)",
            "background": "rgba(0, 0, 0, 0.3)",
            "height": "100%",
            "opacity": 0,
            "position": "fixed",
            "transform": "translate3d(100%, 0, 0)",
            "transition": "opacity 0.3s, transform 0s 0.3s",
            "width": "100%",
            "zIndex": 1000,
          }
        }
      />
      <div
        style={Object {}}
      >
        <div
          className="bm-burger-button"
          style={
            Object {
              "zIndex": 1000,
            }
          }
        >
          <button
            id="react-burger-menu-btn"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "cursor": "pointer",
                "fontSize": 0,
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
                "zIndex": 1,
              }
            }
          >
            Open Menu
          </button>
          <span>
            <span
              className="bm-burger-bars"
              style={
                Object {
                  "height": "20%",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": "0%",
                }
              }
            />
            <span
              className="bm-burger-bars"
              style={
                Object {
                  "height": "20%",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": "40%",
                }
              }
            />
            <span
              className="bm-burger-bars"
              style={
                Object {
                  "height": "20%",
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": "80%",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="bm-menu-wrap"
        id=""
        style={
          Object {
            "MozTransform": "translate3d(100%, 0, 0)",
            "MsTransform": "translate3d(100%, 0, 0)",
            "OTransform": "translate3d(100%, 0, 0)",
            "WebkitTransform": "translate3d(100%, 0, 0)",
            "height": "100%",
            "position": "fixed",
            "right": 0,
            "transform": "translate3d(100%, 0, 0)",
            "transition": "all 0.5s",
            "width": "100%",
            "zIndex": 1100,
          }
        }
      >
        <div
          className="bm-menu"
          style={
            Object {
              "boxSizing": "border-box",
              "height": "100%",
              "overflow": "auto",
            }
          }
        >
          <nav
            className="bm-item-list"
            style={
              Object {
                "height": "100%",
              }
            }
          >
            <li
              className="styles__Item-sc-2csmix-4 jVPdVz bm-item"
              style={
                Object {
                  "display": "block",
                }
              }
              tabIndex={-1}
            >
              <a
                className="menu-item"
                href="/rockets"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Rockets
              </a>
            </li>
            <li
              className="styles__Item-sc-2csmix-4 jVPdVz bm-item"
              style={
                Object {
                  "display": "block",
                }
              }
              tabIndex={-1}
            >
              <a
                className="menu-item"
                href="/launches"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Launches
              </a>
            </li>
            <li
              className="styles__Item-sc-2csmix-4 jVPdVz bm-item"
              style={
                Object {
                  "display": "block",
                }
              }
              tabIndex={-1}
            >
              <a
                className="menu-item"
                href="/capsules"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Capsules
              </a>
            </li>
            <li
              className="styles__Item-sc-2csmix-4 jVPdVz bm-item"
              style={
                Object {
                  "display": "block",
                }
              }
              tabIndex={-1}
            >
              <a
                className="menu-item"
                href="/stats"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Stats
              </a>
            </li>
            <li
              className="styles__Item-sc-2csmix-4 jVPdVz bm-item"
              style={
                Object {
                  "display": "block",
                }
              }
              tabIndex={-1}
            >
              <a
                className="menu-item"
                href="/about"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                About
              </a>
            </li>
          </nav>
        </div>
        <div
          style={Object {}}
        >
          <div
            className="bm-cross-button"
            style={
              Object {
                "height": 24,
                "position": "absolute",
                "right": 8,
                "top": 8,
                "width": 24,
              }
            }
          >
            <button
              id="react-burger-cross-btn"
              onClick={[Function]}
              style={
                Object {
                  "background": "transparent",
                  "border": "none",
                  "cursor": "pointer",
                  "fontSize": 0,
                  "height": "100%",
                  "left": 0,
                  "margin": 0,
                  "padding": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                  "zIndex": 1,
                }
              }
              tabIndex={-1}
            >
              Close Menu
            </button>
            <span
              style={
                Object {
                  "position": "absolute",
                  "right": "14px",
                  "top": "6px",
                }
              }
            >
              <span
                className="bm-cross"
                style={
                  Object {
                    "height": 14,
                    "position": "absolute",
                    "transform": "rotate(45deg)",
                    "width": 3,
                  }
                }
              />
              <span
                className="bm-cross"
                style={
                  Object {
                    "height": 14,
                    "position": "absolute",
                    "transform": "rotate(-45deg)",
                    "width": 3,
                  }
                }
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </ul>
</header>
`;
